Zeus w caam review (#2)
bgnetworks/meta-bgn-essa: docs/Quick_Start_Guide.md

> The files and directories that are in the bgn-essa-c444

Reviewer: ```suggestion
Files and directories that are in the `bgn-essa-c444`:
```

<p align="left">
    <img src="assets/sync_dir.png" width=600 alt="sync_dir" />
</p>

> The Yocto meta directories that are in the sources

Reviewer: ```suggestion
Yocto meta directories that are in `sources`:
```

<p align="left">
    <img src="assets/yocto_meta_dir.png" width=800 alt="yocto_meta_dir" />
</p>

> #### 4. setup the build environment at first time

Reviewer: I think we should start each numbered step with a capital letter.

The following build configurations should show up in the terminal:

<p align="left">
    <img src="assets/yocto_build_config.png" width=800 alt="yocto_build_config" />
</p>

> **Note**: _the image build may take a few hours at the first time_

Reviewer: ```suggestion
**Note**: _The initial image build may take a few hours_
```

<p align="left">
    <img src="assets/dmsetup_command.png" width=800 alt="dmsetup_command" />
</p>

#### 9. Build file system

```bash
mkfs.ext4 /dev/mapper/myEncryptedBlock
```

#### 10. Mount the encrypted block

```bash
mkdir -p /mnt/myBlock
mount /dev/mapper/myEncryptedBlock /mnt/myBlock
```

#### 11. Create a new file in the encrypted block

```bash
echo "This is a test of disk encryption on i.MX" > /mnt/myBlock/readme.txt
```

#### 12. Umount and remove the encrypted block

```bash
umount /mnt/myBlock
dmsetup remove myEncryptedBlock
```

#### 13. Power cycle the `itx-p-c444` board and log in as `root` after the booting is completed

#### 14. Import the generated random key

```bash
cd /data/caam
caam-keygen import mykey.bb importKey
```

<p align="left">
    <img src="assets/caam_importkey.png" width=400 alt="caam_importkey" />
</p>

#### 15. Add the key into the Linux keyring

```bash
cat importKey | keyctl padd logon mykey2: @s
```

<p align="left">
    <img src="assets/linux_keyring2.png" width=500 alt="linux_keyring2" />
</p>

#### 16. Link the file to loop device

```bash
losetup /dev/loop0 encrypted.img
```

#### 17. Use the imported key for block encryption

```bash

> dmsetup -v create myEncryptedBlock --table "0 $(blockdev --getsz /dev/loop0) crypt capi:tk(cbc(aes))-plain :36:logon:mykey2: 0 /dev/loop0 0 1 sector_size:512"

Reviewer 2: Nice catch and kudos for updating the `dmsetup` commands 👍
